03.bowling/bowling.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+score = ARGV[0]
+scores = score.split(',')
+shots = []
+scores.each do |s|
+  if s == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << s.to_i
+  end
+end
+
+frames = []
+shots.each_slice(2) do |s|
+  frames << s
+end
+
+point = 0
+frames.each.with_index do |frame, i|
+  point += frame.sum
+  next if frame.sum != 10 || i >= 9
+
+  point += shots[(i + 1) * 2]
+  point += shots[(i + 2) * 2] if frame[0] == 10 && shots[(i + 1) * 2] == 10 # ストライクの次がストライクの場合
+  point += shots[(i + 1) * 2 + 1] if frame[0] == 10 # 上記以外のストライクの場合
```

Comment:
- 早期リターンをさらに使ってもいいかもしれません。
- 条件分岐は局所的なほうがわかりやすいです。

だいぶささいな指摘になってきたので、コードをそのまま書きますが、

```suggestion
  point += shots[(i + 1) * 2]
  next if frame[0] != 10
  
  next_next_shot_index =  shots[(i + 1) * 2] == 10 ? (i + 2) * 2 : (i + 1) * 2 + 1
  point += shots[next_next_shot_index]
```

でどうでしょうか。